ipu-plugin/pkg/ipuplugin/lifecycleservice.go

```diff
@@ -230,7 +236,227 @@ func configureChannel(mode, daemonHostIp, daemonIpuIp string) error {
+	}
+	defer srcFile.Close()
+
+	// Create the destination file on the remote server.
+	remoteFilePath := "/work/scripts/linux_networking.pkg"
+	dstFile, err := sftpClient.Create(remoteFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to create remote file: %s", err)
+	}
+	defer dstFile.Close()
+
+	// Copy the file contents to the destination file.
+	_, err = io.Copy(dstFile, srcFile)
+	if err != nil {
+		return fmt.Errorf("failed to copy file: %s", err)
+	}
+
+	// Ensure that the file is written to the remote filesystem.
+	err = dstFile.Sync()
+	if err != nil {
+		return fmt.Errorf("failed to sync file: %s", err)
+	}
+
+	// Start a session.
+	session, err := client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create session: %s", err)
+	}
+	defer session.Close()
+
+	shellScript := `#!/bin/sh
+	CP_INIT_CFG=/etc/dpcp/cfg/cp_init.cfg
+	echo "Checking for custom package..."
+	if [ -e linux_networking.pkg ]; then
+		echo "Custom package linux_networking.pkg found. Overriding default package"
+		cp  linux_networking.pkg /etc/dpcp/package/
+		rm -rf /etc/dpcp/package/default_pkg.pkg
+		ln -s /etc/dpcp/package/linux_networking.pkg /etc/dpcp/package/default_pkg.pkg
+		sed -i 's/sem_num_pages = 1;/sem_num_pages = 25;/g' $CP_INIT_CFG
+		sed -i 's/acc_apf = 4;/acc_apf = 8;/g' $CP_INIT_CFG
+		sed -i 's/comm_vports = ((\[5,0\],\[4,0\]));/comm_vports = ((\[5,0\],\[4,0\]),(\[0,3\],\[4,2\]));/g' $CP_INIT_CFG
+	else
+		echo "No custom package found. Continuing with default package"
+	fi
+	`
+
+	commands := fmt.Sprintf(`
+	cat <<'EOF' > /work/scripts/load_custom_pkg.sh
+	%s
+	EOF
+	`, shellScript)
+
+	// Run a command on the remote server and capture the output.
+	var stdoutBuf bytes.Buffer
+	session.Stdout = &stdoutBuf
+	err = session.Run(commands)
+	if err != nil {
+		return fmt.Errorf("failed to run commands: %s", err)
+	}
+	fmt.Println(stdoutBuf.String())
+
+	return nil
+}
+
+func (s *LifeCycleServiceServer) countAPFDevices() int {
+
+	sysClassNet := "/sys/class/net"
+	deviceCode := "0x1452"
+
+	files, err := os.ReadDir(sysClassNet)
+	if err != nil {
+		if os.IsNotExist(err) {
+			fmt.Printf("Error: %s\n", err)
+			return 0
+		}
+	}
+
+	count := 0
+	for _, file := range files {
+		deviceCodeByte, err := os.ReadFile(filepath.Join(sysClassNet, file.Name(), "device/device"))
+		if err != nil {
+			fmt.Printf("Error: %s\n", err)
+		}
+
+		device_code := strings.TrimSpace(string(deviceCodeByte))
+
+		if device_code == deviceCode {
+			count++
+		}
+	}
+
+	return count
+}
+
+func (s *LifeCycleServiceServer) formatMAC(mac string) (string, error) {
+	if len(mac) != 12 {
+		return "", fmt.Errorf("invalid MAC address length")
+	}
+
+	var pairs []string
+	for i := 0; i < len(mac); i += 2 {
+		pairs = append(pairs, mac[i:i+2])
+	}
+
+	formattedMAC := strings.Join(pairs, ":")
+
+	return formattedMAC, nil
+}
+
+func (s *LifeCycleServiceServer) checkMAC() (bool, string) {
+	cmd := exec.Command("sh", "-c", "dmidecode | sha256sum | head -c 12")
```

Comment:
this functionality needs to change once we decide how to do this.

the rest of the code is sound.